source/scripts/task-list.js

```diff
@@ -0,0 +1,249 @@
+class TaskListWidget extends HTMLElement {
+
+    /*
+    A basic constructor that initializes the tasks array.
+    */
+    constructor() {
+        super(); // necessary js call prior to self-referencing via this
+        this.tasks = [];
+    }
+
+    /*
+    A basic init method that sets a basic innerHTML template for the custom element.
+    */
+    init() {
+        this.innerHTML = `
+            <div class="container">
+                <h1>Task List</h1>
+                
+                <div class="inputContainer">
+                    <div>
+                        <input type="text" id="taskInput" placeholder="Enter task...">
+                    </div>
+                    <div>
+                        <select id="priorityInput">
+                            <option value="0">Select one</option>
+                            <option value="4">critical</option>
+                            <option value="3">high</option>
+                            <option value="2">medium</option>
+                            <option value="1">low</option>
+                        </select>
+                    </div>
+                </div>
+                <button id="addBtn">Add Task</button>
+                
+                <div class="list-widget" id="taskList">
+                    <!-- Tasks will be added dynamically here -->
+                </div>
+            </div>`;
+
+    }
+
+    /* 
+    The connectedCallback method is called when the element is added to the DOM.
+    This method fetches the JSON/localStorage data and renders the tasks.
+    */
+    connectedCallback() {
+        // Check if tasks are stored in local storage, if so, load them and terminate function
+        if (localStorage.getItem("tasks") !== null) {
+            this.tasks = JSON.parse(localStorage.getItem("tasks"));
+            this.renderTasks();
+            return;
+        }
+
+        // If no tasks are stored in local storage, fetch tasks from JSON file
+        const src = this.getAttribute('src');
+        if (src) {
+            fetch(src)
+                .then(response => response.json())
+                .then(data => {
+                    this.tasks = data;
+                    this.renderTasks();
+                })
+                .catch(error => {
+                    console.error('Error fetching JSON:', error);
+                });
+        } else {
+            console.error('No src attribute provided.'); // this will log an error if no src attribute is provided, JS error
+        }
+    }
+
+    /*
+    A basic method that adds a task to the task list.
```

Comment:
Please explain what each parameter of this function is in the JSDoc documentation. Here is an example: [param-tag](https://jsdoc.app/tags-param)